frontend/src/main.tsx

```diff
@@ -4,13 +4,24 @@ import './global.css';
 import { BrowserRouter as Router } from 'react-router-dom';
 import AppRoutes from './AppRoutes.tsx';
 import Auth0ProviderNavigate from './auth/Auth0ProviderNavigate.tsx';
+import { QueryClient, QueryClientProvider } from 'react-query';
+
+const queryClient = new QueryClient({
+  defaultOptions: {
+    queries: {
+      refetchOnWindowFocus: false,
+    },
+  },
+});
 
 createRoot(document.getElementById('root')!).render(
   <StrictMode>
     <Router>
-      <Auth0ProviderNavigate>
-        <AppRoutes />
-      </Auth0ProviderNavigate>
+      <QueryClientProvider client={queryClient}>
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/15#issuecomment-2363730468>#51a289</a><span> - 09/20/2024, 01:31 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The QueryClientProvider is placed inside the Router component, which may limit its scope and potentially cause issues with query caching and state management across routes.
</div>
<div id="fix">
<b>Fix:</b> Move the QueryClientProvider outside of the Router component to ensure it wraps the entire application, including the Router.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -17,11 +17,11 @@ const queryClient = new QueryClient({
  createRoot(document.getElementById('root')!).render(
    <StrictMode>
 -    <Router>
 -      <QueryClientProvider client={queryClient}>
 +    <QueryClientProvider client={queryClient}>
 +      <Router>
          <Auth0ProviderNavigate>
            <AppRoutes />
          </Auth0ProviderNavigate>
 -      </QueryClientProvider>
 -    </Router>
 +      </Router>
 +    </QueryClientProvider>
    </StrictMode>
  );
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged